### PR TITLE
bugfix: missing merge some config from image

### PR DESCRIPTION
**Pull request by Ace-Tang**
add user, labels, stop signal and exposed ports config from image
config in create container.

Signed-off-by: Ace-Tang <aceapril@126.com>

### CODE REVIEW

File: daemon/mgr/container_types.go [hunk 1]
```diff
@@ -285,28 +285,63 @@ func (c *Container) StopTimeout() int64 {
...
+	newEnvSlice, err := mergeEnvSlice(c.Config.Env, imageConf.Env)
 	if err != nil {
 		return err
 	}
 	c.Config.Env = newEnvSlice
 	if c.Config.WorkingDir == "" {
-		c.Config.WorkingDir = config.WorkingDir
+		c.Config.WorkingDir = imageConf.WorkingDir
+	}
+
```
Comment (fuweid) [inline daemon/mgr/container_types.go]: missing volumes here?

File: daemon/mgr/container.go [hunk 2]
```diff
@@ -355,6 +355,18 @@ func (mgr *ContainerManager) Create(ctx context.Context, name string, config *ty
...
+		img, err := mgr.Client.GetImage(ctx, config.Image)
+		ociImage, err := containerdImageToOciImage(ctx, img)
+		if err != nil {
+			return ocispec.ImageConfig{}, err
+		}
+		return ociImage.Config, nil
+	}); err != nil {
+		return nil, err
+	}
+
```
Comment (Ace-Tang) [inline daemon/mgr/container.go]: put merge funtion before initContainerStorage, make mount points generate completely.

File: daemon/mgr/container_types.go (hunk 1 above)
Comment (Ace-Tang) [reply daemon/mgr/container_types.go]: Update and add.

File: daemon/mgr/container.go [hunk 3]
```diff
@@ -355,6 +355,18 @@ func (mgr *ContainerManager) Create(ctx context.Context, name string, config *ty
 		HostConfig: config.HostConfig,
 	}
 
+	// merge image's config into container
+	if err := container.merge(func() (ocispec.ImageConfig, error) {
+		img, err := mgr.Client.GetImage(ctx, config.Image)
```
Comment (fuweid) [inline daemon/mgr/container.go]: no error check?

File: daemon/mgr/container_types_test.go [hunk 4]
```diff
@@ -82,3 +86,174 @@ func TestContainer_FormatStatus(t *testing.T) {
...
+			},
+		},
+	} {
+		err := tc.c.merge(func() (v1.ImageConfig, error) {
+			return tc.image, nil
+		})
+		assert.NoError(err)
+
+		// sort slice
+
```
Comment (fuweid) [inline daemon/mgr/container_types_test.go]: remove the empty line

File: daemon/mgr/container.go (hunk 3 above)
Comment (Ace-Tang) [reply daemon/mgr/container.go]: This should ask the one wrote for it, seems like a bug.

**Comment by fuweid** [comment]
ping @YaoZengzeng and @Starnop 

could you help us to check this failure case? https://travis-ci.org/alibaba/pouch/jobs/420765951 

```
• Failure [1.013 seconds]
[k8s.io] Security Context
/home/travis/gopath/src/github.com/kubernetes-incubator/cri-tools/pkg/framework/framework.go:72
  bucket
  /home/travis/gopath/src/github.com/kubernetes-incubator/cri-tools/pkg/validate/security_context.go:280
    runtime should support RunAsUser [It]
    /home/travis/gopath/src/github.com/kubernetes-incubator/cri-tools/pkg/validate/security_context.go:324
    The stdout output of execSync should be 1001
    
    Expected
        <string>: 
    to equal
        <string>: 1001
```

thanks

**Pull request merged**